embed/ephy-fx-password-notification.c

```diff
@@ -0,0 +1,162 @@
+/* -*- Mode: C; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/*
+ *  Copyright © 2016 Gabriel Ivascu <ivascu.gabriel59@gmail.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "config.h"
+#include "ephy-fx-password-notification.h"
+
+#include "ephy-notification-manager.h"
+
+#include <glib/gi18n.h>
+
+struct _EphyFxPasswordNotification {
+  GtkGrid                  parent_instance;
+
+  GtkWidget               *note;
+  GtkWidget               *suggestion;
+
+  const char              *user;
+};
+
+struct _EphyFxPasswordNotificationClass {
+  GtkGridClass parent_class;
+};
+
+enum {
+  PROP_0,
+  PROP_USER
+};
+
+G_DEFINE_TYPE (EphyFxPasswordNotification, ephy_fx_password_notification, GTK_TYPE_GRID);
+
+static void
+ephy_fx_password_notification_constructed (GObject *object)
+{
+  EphyFxPasswordNotification *self = EPHY_FX_PASSWORD_NOTIFICATION (object);
+  char *text;
+
+  text = g_strdup_printf (_("We noticed that the password of your Firefox "
```

Comment:
It's OK, but I prefer to avoid first person here. For example: "The password of your Firefox account <b>%s</b> seems to have been changed." (Note the lowercase 'a'.)

One more thing: we try to avoid including markup in translatable comments, as translators have a habit of breaking it, then we get weird bugs that occur only in particular locales. Unfortunately it's impossible to get rid of format modifiers, but getting rid of markup is only slightly awkward. You just need to add one more `g_strdup_printf`, e.g.:

```
char *account_name;
char * text;

account_name = g_strdup_printf ("<b>%s</b>", self->user);
text = _("We noticed that the password of your Firefox account %s has been changed", account_name);

// g_free them both
```